src/hotspot/share/runtime/mutexLocker.cpp

```diff
@@ -335,7 +335,7 @@ void mutex_init() {
   def(NMethodSweeperStats_lock     , PaddedMutex  , special,     true,  _safepoint_check_never);
   def(ThreadsSMRDelete_lock        , PaddedMonitor, special,     true,  _safepoint_check_never);
   def(ThreadIdTableCreate_lock     , PaddedMutex  , leaf,        false, _safepoint_check_always);
-  def(SharedDecoder_lock           , PaddedMutex  , native,      true,  _safepoint_check_never);
+  def(SharedDecoder_lock           , PaddedMutex  , tty-1,       true,  _safepoint_check_never);
```

Comment:
Sorry I don't understand how this you moved this ranking form being ultra-high to being one of the lowest. What are you actually saying about this lock? What rules are you applying to determine what the rank should be?

Reply:
[not a review, just replying to @dholmes-ora] Mutex::native isn't an "ultra-high" rank, it is outside the ranking system.  Rank checks are not applied for Mutex::native.  See the occurrences of Mutex::rank in mutex.cpp.

Reply:
I stand corrected - thanks @kimbarrett  - but my question remains valid. If this was considered outside the ranking system then it was presumably because a suitable rank could not be determined, so what rules are now being used to determine this new rank?

Reply:
It was outside the ranking system because no appropriate ranking was found for it at the time and 'native' was available.  There's no evidence that this lock should be outside the ranking system.  The ranking system is to help provide deadlock detection and this lock isn't that special.  It is a low ranked lock however because it's taken with the tty lock held.

Reply:
The SharedDecoder lock is used during crash reporting and so could potentially be taken whilst any other lock is held. While we special case the use of the Decoder in the initial error reporting thread, by not taking the lock due to the deadlock potential, it will still be taken by threads that encounter a secondary error. That is what makes this lock "special".

Reply:
That's why it's ranking is very low, below tty.  If we think it should not participate in lock ranking for deadlock detection we can make it a PlatformMutex, but I don't see any evidence that this is needed.

Reply:
I took a closer look at this one and I was wrong about its use. The decoder lock won't be taken when a secondary error occurs as we would not reach any code that tries to use it in that case.